server/motr_common.cc

```diff
@@ -163,36 +163,27 @@ int create_new_instance_id(struct m0_uint128 *ufid) {
   return rc;
 }
 
-void teardown_motr_op(struct m0_op *op) {
-  if (op != NULL) {
-    if (op->op_sm.sm_state == M0_OS_LAUNCHED) {
-      m0_op_cancel(&op, 1);
-    }
-    if (op->op_sm.sm_state == M0_OS_INITIALISED ||
-        op->op_sm.sm_state == M0_OS_STABLE ||
-        op->op_sm.sm_state == M0_OS_FAILED) {
-      m0_op_fini(op);
-    }
-    if (op->op_sm.sm_state == M0_OS_UNINITIALISED) {
-      m0_op_free(op);
-    }
-  }
-}
+void teardown_motr_op(struct m0_op *op) { teardown_motr_cancel_wait_op(op, 0); }
 
-void teardown_motr_cancel_wait_op(struct m0_op *op) {
-  if (op != NULL) {
-    if (op->op_sm.sm_state == M0_OS_LAUNCHED) {
+void teardown_motr_cancel_wait_op(struct m0_op *op, int sec) {
+  if (!op) return;
+
+  switch (op->op_sm.sm_state) {
+    case M0_OS_LAUNCHED:
+    case M0_OS_EXECUTED:
```

Comment:
This (M0_OS_EXECUTED: ). was not there before

Reply:
I have been reading comments in .../cortx-s3server/third_party/motr/motr/client.h much time and think that M0_OS_EXECUTED is needed

Reply:
@madhavemuri is M0_OS_EXECUTED needed ?

Reply:
M0_OS_EXECUTED is equivalent to  M0_OS_STABLE so not needed

Reply:
M0_OS_EXECUTED has been removed.